src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionRepeatDbHandler.java

```diff
@@ -55,139 +53,129 @@
-					LOGGER.error(e.getMessage());
-					e.printStackTrace();
-				} 
-			}
-		}
-		LOGGER.info("AggrObjId not found for SubscriptionId in SubscriptionRepeatFlagHandlerDb -> Returning FALSE.");
-		return false;
-	}
-	
+
+    public boolean checkIfAggrObjIdExistInSubscriptionAggrIdsMatchedList(String subscriptionId, int requirementId,
+            String aggrObjId) {
+
+        LOGGER.debug("Checking if AggrObjId: " + aggrObjId + " exist in SubscriptionId: " + subscriptionId
+                + " AggrId matched list.");
+        String subscriptionQuery = "{\"subscriptionId\" : \"" + subscriptionId + "\"}";
+        List<String> objArray = mongoDbHandler.find(dataBaseName, collectionName, subscriptionQuery);
+        if (objArray != null && !objArray.isEmpty()) {
+            JsonNode jNode = null;
+            LOGGER.debug("Making AggrObjId checks on SubscriptionId document: " + objArray.get(0));
+            try {
+                jNode = mapper.readTree(objArray.get(0));
+            } catch (Exception e) {
+                LOGGER.error(e.getMessage());
+                e.printStackTrace();
+            }
+            if (jNode.get("subscriptionId").asText().trim().equals(subscriptionId)) {
+                LOGGER.debug("SubscriptionId \"", subscriptionId,
+                        "\" , exist in document. Checking if AggrObjId has matched earlier.");
+                List<String> listAggrObjIds = null;
+                LOGGER.debug("Subscription requirementId: " + requirementId + " and Requirements content:\n"
+                        + jNode.get("requirements").get(new Integer(requirementId).toString()));
+                try {
+                    ObjectReader reader = mapper.readerFor(new TypeReference<List<String>>() {
+                    });
+                    JsonNode ids = jNode.get("requirements").get("" + requirementId);
+                    if (ids == null)
+                        return false;
+                    listAggrObjIds = reader.readValue(ids);
+
+                    if (requirementId > (listAggrObjIds.size() - 1)) {
+                        LOGGER.debug("RequirementId: " + requirementId + " and SubscriptionId: " + subscriptionId
+                                + "\nhas not matched any AggregatedObject yet. No need to do anymore check.");
+                        return false;
+                    }
+                    String idStr = listAggrObjIds.get(requirementId);
+                    if (idStr.equals(aggrObjId)) {
+                        LOGGER.debug("Subscription has matched aggrObjId already: " + aggrObjId);
+                        return true;
+                    }
+                } catch (Exception e) {
+                    LOGGER.error(e.getMessage());
+                    e.printStackTrace();
+                }
+            }
+        }
+        LOGGER.debug("AggrObjId not found for SubscriptionId in SubscriptionRepeatFlagHandlerDb -> Returning FALSE.");
+        return false;
+    }
```

Comment:
I think this method could be broken up into smaller methods to make it more readable.

Reply:
The method is not that big, just extensive logging and spaces. That makes the method look big.

Logging could be cleaned up, for example e.printStackTrace should be removed.